src/main/java/com/google/cloud/tools/appengine/operations/DevServer.java

```diff
@@ -82,14 +82,15 @@ public void run(RunConfiguration config) throws AppEngineException {
       jvmArguments.addAll(config.getJvmFlags());
     }
 
-    if (!JAVA_SPECIFICATION_VERSION.value().equals("1.8")) {
-      // Due to JPMS restrictions, Java11 or later need more flags:
-      jvmArguments.add("--add-opens");
-      jvmArguments.add("java.base/java.net=ALL-UNNAMED");
-      jvmArguments.add("--add-opens");
-      jvmArguments.add("java.base/sun.net.www.protocol.http=ALL-UNNAMED");
-      jvmArguments.add("--add-opens");
-      jvmArguments.add("java.base/sun.net.www.protocol.https=ALL-UNNAMED");
+    // Check if the RunConfiguration has the Project JDK Version defined first
+    // The custom value takes priority over the System Property
+    String jdkVersionString = config.getProjectJdkVersion();
+    if (jdkVersionString == null) {
+      jdkVersionString = JAVA_SPECIFICATION_VERSION.value();
+    }
+    int jdkVersion = getJdkVersion(jdkVersionString);
+    if (jdkVersion > 8) {
+      addJpmsRestrictionArguments(jvmArguments);
```

Comment:
If this library uses logging, would you insert debug log about the value `jdkVersionString` and `jdkVersion` before line 92?

Reply:
Done